src/js/component.js

```diff
@@ -220,28 +220,28 @@ class Component {
    * @return {string}
    *         The localized string or if no localization exists the english string.
    */
-  localize(string) {
+  localize(string, tokens) {
     const code = this.player_.language && this.player_.language();
     const languages = this.player_.languages && this.player_.languages();
+    const language = languages && languages[code];
+    const primaryCode = code && code.split('-')[0];
+    const primaryLang = languages && languages[primaryCode];
 
-    if (!code || !languages) {
-      return string;
-    }
-
-    const language = languages[code];
+    let localizedString = string;
 
     if (language && language[string]) {
-      return language[string];
+      localizedString = language[string];
+    } else if (primaryLang && primaryLang[string]) {
+      localizedString = primaryLang[string];
     }
 
-    const primaryCode = code.split('-')[0];
-    const primaryLang = languages[primaryCode];
-
-    if (primaryLang && primaryLang[string]) {
-      return primaryLang[string];
+    if (tokens) {
+      localizedString = localizedString.replace(/\{(\d+)\}/g, function(match, index) {
+        return tokens[index - 1];
```

Comment:
To elaborate and update (after some discussion with @OwenEdwards).
The key should be `progress bar timing: currentTime={1}, duration={2}` and the value would be `{1} of {2}`. That way, as you're reading the JSON files, you have some idea of what the items are.
Then, we would want to add a 3rd argument to `localize`, a defaults argument, which would be `{1} of {2}` which is what should be used in the case a language file isn't included (and so we don't have to require en.json to be included).